aggregator_core/src/datastore.rs

```diff
@@ -4278,9 +4336,26 @@ pub mod models {
     {
     }
 
+    impl<const SEED_SIZE: usize, Q: QueryType, A: vdaf::Aggregator<SEED_SIZE, 16>> Hash
+        for AggregationJob<SEED_SIZE, Q, A>
```

Comment:
Surprisingly, this worked -- deriving `[Partial]Eq` had some problem with generic structs, so I assumed `Hash` worked the same way. I should take a look (outside this PR) and see if we can get rid of all these manual `[Partial]Eq` implementations using a similar technique, since they risk a similar error on introduction of a new field.

Reply:
It turned out this only worked because nothing was actually using the `AggregationJob` `Hash` implementation, so the funky stuff around the derived hash implementation requiring the type parameters to implement `Hash` as well didn't matter.

Instead, I just removed `AggregationJob`'s `Hash` implementation entirely.